recipes/gtsam/all/conanfile.py

```diff
@@ -75,43 +120,91 @@ def export_sources(self):
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
+        if Version(self.version) >= "4.1":
+            del self.options.build_wrap
+            del self.options.install_cython_toolbox
+            del self.options.typedef_points_to_vectors
+            del self.options.wrap_serialization
+        else:
+            del self.options.slow_but_correct_betweenfactor
 
     def configure(self):
         if self.options.shared:
             self.options.rm_safe("fPIC")
         if self.options.with_TBB:
             self.options["onetbb"].tbbmalloc = True
+        if not self.options.support_nested_dissection:
+            self.options.rm_safe("use_vendored_metis")
+        elif not self.options.use_vendored_metis:
+            # GTSAM expects 32-bit integers for Metis
+            self.options["metis"].with_64bit_types = False
```

Comment:
Could you post logs of this configuration building correctly please :)? CCI's CI won't get to try these combinations, and it would be great to have some proof for future maintenance which shows that some of the new options were tested properly

Reply:
Hmm, it no longer builds with the new `metis` recipe, unfortunately. I'm reverting that feature for now and will possibly add it in a future PR instead.

Reply:
Hi @valgur! Is this happening because of the latest Metis recipe version? Could we fix that before removing that option?

Reply:
I think so, yes. I intend to look into the Metis issue, but I would leave that for a separate PR, since this one is quite comprehensive already. It's not affecting any existing functionality in the GTSAM recipe anyway.